waku/protocol/v2/filter.nim

```diff
@@ -1,9 +1,11 @@
+import libp2p/protocols/pubsub/rpc/messages
+
 import 
   tables
 
 type 
 
-  FilterMessageHandler* = proc(msg: seq[byte]) {.gcsafe, closure.}
+  FilterMessageHandler* = proc(msg: Message) {.gcsafe, closure.}
```

Comment:
how come this isn't Topichandler or ContentHandler per https://github.com/status-im/nim-waku/blob/master/docs/api/v2/node.md? What is Filter and notify and do we need to add new names?

Reply:
@oskarth, we could name it that, but filters are more for internode communication. Like various components within the node, so I am not sure if those match.

Reply:
What is internode communication? Either it is part of the protocol, the node API or it is an implementation detail to achieve one or the other.

Reply:
its an implementation detail to be able to save messages recieved to the mailserver. Not sure how you'd want to put that

Reply:
There is no mailserver in Waku v2 but fine, if implementation detail lets merge for now